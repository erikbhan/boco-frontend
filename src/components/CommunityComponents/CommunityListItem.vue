<template>
  <NotificationModal
    @close="this.dialogOpen = false"
    :visible="dialogOpen"
    :title="community.name"
    :message="community.description"
  >
  </NotificationModal>
  <div
    @click="toggleDialog()"
    class="bg-white shadow dark:bg-gray-800 select-none cursor-pointer hover:bg-gray-50 flex items-center p-4"
  >
    <div class="h-10 w-10 flex flex-col justify-center items-center mr-4">
      <img alt="groupIMG" src="../../assets/group.png" />
    </div>
    <div class="flex-1 pl-1 overflow-hidden">
      <div class="font-medium dark:text-white truncate">
        {{ community.name }}
      </div>
    </div>
    <div class="flex flex-row justify-center items-center">
      <LockClosedIcon
        v-if="community.visibility === 0"
        class="max-h-6 max-w-6 shrink m-2"
      />
      <LockOpenIcon
        v-if="community.visibility === 1"
        class="max-h-6 max-w-6 shrink m-2"
      />
      <button
        @click="goToJoin(community.communityId)"
        v-if="!member"
        class="px-4 py-2 w-24 font-medium tracking-wide text-white capitalize transition-colors duration-200 transform bg-blue-600 rounded-md hover:bg-blue-500 focus:outline-none focus:ring focus:ring-blue-300 focus:ring-opacity-80"
      >
        Bli med
      </button>
      <button
        v-if="member"
        @click="goToGroup(community.communityId)"
        class="px-4 py-2 w-24 font-medium tracking-wide text-white capitalize transition-colors duration-200 transform bg-blue-600 rounded-md hover:bg-blue-500 focus:outline-none focus:ring focus:ring-blue-300 focus:ring-opacity-80"
      >
        GÃ¥ til
      </button>
    </div>
  </div>
</template>

<script>
import { LockClosedIcon, LockOpenIcon } from "@heroicons/vue/outline";
import NotificationModal from "@/components/BaseComponents/NotificationModal.vue";

export default {
  name: "CommunityListItem",
  components: {
    NotificationModal,
    LockClosedIcon,
    LockOpenIcon,
  },
  data() {
    return {
      dialogOpen: false,
    };
  },
  props: {
    community: Object,
    member: Boolean,
  },
  methods: {
    goToGroup(id) {
      this.$router.push("/community/" + id);
    },
    goToJoin(id) {
      this.$router.push("/community/" + id + "/join");
    },
    toggleDialog() {
      this.dialogOpen = !this.dialogOpen;
    },
  },
};
</script>
