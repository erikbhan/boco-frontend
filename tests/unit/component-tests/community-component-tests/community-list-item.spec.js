import { mount } from "@vue/test-utils";
import CommunityListItem from "@/components/CommunityComponents/CommunityListItem.vue"

describe("CommunityListItem component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(CommunityListItem, {
        //passing prop to component
        props: {
            group: {
                communityId: 0,
                name: "string",
                description: "string",
                visibility: 0,
                location: "string",
                picture: "string"
            }
        }
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});