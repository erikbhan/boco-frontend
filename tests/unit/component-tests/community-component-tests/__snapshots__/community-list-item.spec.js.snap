// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommunityListItem component renders correctly 1`] = `
<div
  data-v-app=""
>
  
  
  <!-- Main modal -->
  <!--v-if-->
  
  <div
    class="bg-white shadow dark:bg-gray-800 select-none cursor-pointer hover:bg-gray-50 flex items-center p-4"
  >
    <div
      class="h-10 w-10 flex flex-col justify-center items-center mr-4"
    >
      <svg
        aria-hidden="true"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17 20h5v-2a3 3 0 00-5.356-1.857M17 20H7m10 0v-2c0-.656-.126-1.283-.356-1.857M7 20H2v-2a3 3 0 015.356-1.857M7 20v-2c0-.656.126-1.283.356-1.857m0 0a5.002 5.002 0 019.288 0M15 7a3 3 0 11-6 0 3 3 0 016 0zm6 3a2 2 0 11-4 0 2 2 0 014 0zM7 10a2 2 0 11-4 0 2 2 0 014 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <!-- TODO: USE COMMUNITY IMAGE &lt;img alt="Felleskapets bilde" src="@/assets/group.png" /&gt; -->
    </div>
    <div
      class="flex-1 pl-1 overflow-hidden"
    >
      <div
        class="font-medium dark:text-white truncate"
      >
        string
      </div>
    </div>
    <div
      class="flex flex-row justify-center items-center"
    >
      <svg
        aria-hidden="true"
        class="max-h-6 max-w-6 shrink m-2"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 15v2m-6 4h12a2 2 0 002-2v-6a2 2 0 00-2-2H6a2 2 0 00-2 2v6a2 2 0 002 2zm10-10V7a4 4 0 00-8 0v4h8z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </div>
  </div>
  
</div>
`;
